openedx/features/badging/models.py

```diff
@@ -120,13 +128,16 @@ def assign_badge(cls, user_id, badge_id, community_id):
                 raise Exception(error)
 
             all_team_members = CourseTeamMembership.objects.filter(team_id=team_group_chat['team_id'])
+            assigned = False
             for member in all_team_members:
-                UserBadge.objects.get_or_create(
+                user_badge, assigned = UserBadge.objects.get_or_create(
                     user_id=member.user_id,
                     badge_id=badge_id,
                     course_id=course_id,
                     community_id=community_id
                 )
+
+            return assigned
```

Comment:
Actually this was the reason, I was saying I ll be sending user_ids(to NodeBB) to whom badge is assigned after this call, So that only those users will get congrats popup who actually earned the badge after this call.
But according to our discussion during sprint planning, "All team members will have assigned the badge if meet threshold. As we also handled the missing badge assignment on team joining."

Reply:
Let's have a discussion on this.